crates/extrinsics/src/lib.rs

```diff
@@ -303,6 +303,44 @@ pub fn parse_code_hash(input: &str) -> Result<<DefaultConfig as Config>::Hash> {
     Ok(arr.into())
 }
 
+/// Calculate the total contract deposit.
+async fn get_contract_total_deposit(
+    storage_base_deposit: Balance,
+    storage_item_deposit: Balance,
+    storage_byte_deposit: Balance,
+    client: &Client,
+) -> Result<Balance> {
+    let contract_pallet_version = fetch_contracts_pallet_version(client).await?;
+    let mut contract_deposit = storage_base_deposit
+        .saturating_add(storage_item_deposit)
+        .saturating_add(storage_byte_deposit);
+
+    // From contracts pallet version 10 deposit calculation has changed.
+    if contract_pallet_version >= 10 {
+        let existential_deposit_address =
+            api::constants().balances().existential_deposit();
+        let existential_deposit = client.constants().at(&existential_deposit_address)?;
+        contract_deposit = contract_deposit.saturating_sub(existential_deposit);
+    }
+    Ok(contract_deposit)
+}
+
+/// Fetch the contracts pallet version from the storage using the provided client.
+async fn fetch_contracts_pallet_version(client: &Client) -> Result<u16> {
+    let hash_pallet = hashing::twox_128(b"Contracts");
+    let hash_version = hashing::twox_128(b":__STORAGE_VERSION__:");
+    let key = [hash_pallet, hash_version].concat();
+
+    let version = client
+        .rpc()
+        .storage(key.as_slice(), None)
```

Comment:
Raw access to the storage

Reply:
Isn't there an RPC method to fetch a pallet version?

Reply:
I have not found it